tsdb/compact.go

```diff
@@ -58,9 +58,9 @@ type Compactor interface {
 	// Results returned when compactions are in progress are undefined.
 	Plan(dir string) ([]string, error)
 
-	// Write persists a Block into a directory.
-	// No Block is written when resulting Block has 0 samples, and returns empty ulid.ULID{}.
-	Write(dest string, b BlockReader, mint, maxt int64, base *BlockMeta) (ulid.ULID, error)
+	// Write persists one or more Blocks into a directory.
+	// When a resulting Block has 0 samples then its result will be an empty ulid.ULID{}.
```

Comment:
When there is no block produced, it would be a good design to return an empty slice instead of a empty ulid.
```suggestion
	// Write persists one or more Blocks into a directory.
	// No Block is written when resulting Block has 0 samples and returns an empty slice.
	// Prometheus always return one or no block. The interface allows returning more than one
	// block for downstream users to experiment with compactor.
```

Reply:
Let's have the `// Prometheus always return one or no block. The interface allows returning more than one block for downstream users to experiment with compactor.` comment for Compact() as well once you make the change.